src/test/java/org/fcrepo/client/ConnectionManagementIT.java

```diff
@@ -0,0 +1,371 @@
+ *
+ * @author esm
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ConnectionManagementIT {
+
+    /**
+     * Starts a mock HTTP server on a free port
+     */
+    @Rule
+    public MockServerRule mockServerRule = new MockServerRule(this);
+
+    // Set by the above @Rule, initialized on @Before via MockHttpExpectations
+    private MockServerClient mockServerClient;
+
+    /**
+     * URIs that our Mock HTTP server responds to.
+     */
+    private MockHttpExpectations.SupportedUris uris;
+
+    /**
+     * Verifications for HttpClientConnectionManager
+     */
+    static class Verifications {
+
+        /**
+         * Verifies that the expected number of connections have been requested and then closed.
+         *
+         * @param connectionCount the number of connections that have been requested and closed.
+         * @return a Consumer that verifies the supplied HttpClientConnectionManager has opened and closed the expected
+         * number of connections.
+         */
+        static Consumer<HttpClientConnectionManager> connectionRequestedAndClosed(final int connectionCount) {
+            return (connectionManager) -> {
+                // A new connection was requested by the Http client ...
+                verify(connectionManager, times(connectionCount)).requestConnection(any(HttpRoute.class), any());
+
+                // Verify it was released.
+                verify(connectionManager, times(connectionCount)).
+                        releaseConnection(any(HttpClientConnection.class), any(), anyLong(), any(TimeUnit.class));
+            };
+        }
+
+        /**
+         * Verifies that the expected number of connections have been requested and <em>have not been</em> closed.
+         *
+         * @param connectionCount the number of connections that have been requested.
+         * @return a Consumer that verifies the supplied HttpClientConnectionManager has opened the expected
+         * number of connections, but hasn't closed <em>any</em> of them
+         */
+
+        static Consumer<HttpClientConnectionManager> connectionRequestedButNotClosed(final int connectionCount) {
+            return (connectionManager) -> {
+                // A new connection was requested by the Http client ...
+                verify(connectionManager, times(connectionCount)).requestConnection(any(HttpRoute.class), any());
+
+                // Verify it was NOT released.
+                verify(connectionManager, times(0)).
+                        releaseConnection(any(HttpClientConnection.class), any(), anyLong(), any(TimeUnit.class));
+            };
+        }
+
+    }
+
+    /**
+     * FcrepoResponse handlers.
+     */
+    static class FcrepoResponseHandler {
+
+        /**
+         * Closes the InputStream that constitutes the response body.
+         */
+        static Consumer<FcrepoResponse> closeEntityBody = (response) ->
```

Comment:
The `Consumer` is a uniform interface and semantics for the finally block in `connect(...)`.  Tests can pass in the appropriate Consumer for their needs.